### PR TITLE
Add a few new fields

### PR DESCRIPTION
**Pull request by JasonHomely**
Added the following:

- Authority
- yearBuilt
- yearLastRenovated
- subNumber
- lotNumber
- depositTaken

Post merginging steps:
- [ ] Create a new issue to merge `ISalePricing` and `ISaleDetails`

### CODE REVIEW

File: src/OpenRealEstate.Core/Address.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/Address.cs:1]: nitpic: This needed?

File: src/OpenRealEstate.Core/AuthorityType.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/AuthorityType.cs:3]: nitpic: need these, also?

File: src/OpenRealEstate.Core/ISaleDetails.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/ISaleDetails.cs:3]: nitpic

File: src/OpenRealEstate.Core/ISaleDetails.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenRealEstate.Core
+{
+    public interface ISaleDetails
+    {
+        string YearBuilt { get; set; }
+
+        string YearLastRenovated { get; set; }
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/ISaleDetails.cs:11]: What are some values here? could this be an `int` instead?

Also possible to make it a `DateOnly` but I think that's an overkill.

File: src/OpenRealEstate.Core/ISaleDetails.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenRealEstate.Core
+{
+    public interface ISaleDetails
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/ISaleDetails.cs:7]: I'm also thinking that we might removed `ISalePricing` and move the sale pricing into this interface.

Scope for another issue, post this IMO (and i'll do it).

File: src/OpenRealEstate.Core/Rental/RentalListing.cs [hunk 6]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/Rental/RentalListing.cs:1]: Prolly can revert all the changes in this file.

File: src/OpenRealEstate.Core/Rental/RentalPricing.cs [hunk 7]
```diff
@@ -14,6 +14,8 @@ public class RentalPricing
 
         public int? Bond { get; set; }
 
+        public bool DepositTaken { get; set; }
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/Rental/RentalPricing.cs:17]: nitpic: rename to `HasReceivedDeposit`

context: for `bool`'s, I usually prepend it with either `Is` or `Has` etc

File: src/OpenRealEstate.Core/Residential/ResidentialListing.cs [hunk 8]
```diff
@@ -1,8 +1,8 @@
-﻿using System;
+using System;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/Residential/ResidentialListing.cs:1]: nitpic

File: src/OpenRealEstate.Core/Rural/RuralListing.cs [hunk 9]
```diff
@@ -1,8 +1,8 @@
-﻿using System;
+using System;
```
Comment (PureKrome) [inline src/OpenRealEstate.Core/Rural/RuralListing.cs:1]: nitpic

**Review by PureKrome — CHANGES_REQUESTED** [review]

File: src/OpenRealEstate.Core/Address.cs (hunk 1 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Address.cs:1]: system is for exception. i dont know why its coming up as a change it was always there

File: src/OpenRealEstate.Core/AuthorityType.cs (hunk 2 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/AuthorityType.cs:3]: removed

File: src/OpenRealEstate.Core/ISaleDetails.cs (hunk 3 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/ISaleDetails.cs:3]: fixed

File: src/OpenRealEstate.Core/ISaleDetails.cs (hunk 4 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/ISaleDetails.cs:11]: they are a numeric year value, could use int but i have an aversion to that, but your call

File: src/OpenRealEstate.Core/Rental/RentalListing.cs (hunk 6 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Rental/RentalListing.cs:1]: done

File: src/OpenRealEstate.Core/Rental/RentalPricing.cs (hunk 7 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Rental/RentalPricing.cs:17]: fixed

File: src/OpenRealEstate.Core/Residential/ResidentialListing.cs (hunk 8 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Residential/ResidentialListing.cs:1]: no actual change here, i dont know what is going on

File: src/OpenRealEstate.Core/Rural/RuralListing.cs (hunk 9 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Rural/RuralListing.cs:1]: still not change

File: src/OpenRealEstate.Core/Address.cs (hunk 1 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/Address.cs:1]: Ok - so where's `Exception` in this class?

File: src/OpenRealEstate.Core/ISaleDetails.cs (hunk 4 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/ISaleDetails.cs:7]: `int` please. I don't want to see a value of `hello`. etc

File: src/OpenRealEstate.Core/Residential/ResidentialListing.cs (hunk 8 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/Residential/ResidentialListing.cs:1]: Ok - is it needed though?

File: src/OpenRealEstate.Core/Rural/RuralListing.cs (hunk 9 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/Rural/RuralListing.cs:1]: Needed?

**Review by PureKrome — CHANGES_REQUESTED** [review]

**Comment by PureKrome** [comment]
Also @JasonHomely - Codecov highlighted an interesting point -> sub and lot are not getting used anywhere?

the `Address` class has some methods for printing out the address. These need to be included IMO, in these methods.

File: src/OpenRealEstate.Core/Address.cs (hunk 1 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Address.cs:1]: line 183

File: src/OpenRealEstate.Core/ISaleDetails.cs (hunk 4 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/ISaleDetails.cs:7]: updated

File: src/OpenRealEstate.Core/Residential/ResidentialListing.cs (hunk 8 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Residential/ResidentialListing.cs:1]: datetime

File: src/OpenRealEstate.Core/Rural/RuralListing.cs (hunk 9 above)
Comment (JasonHomely) [reply src/OpenRealEstate.Core/Rural/RuralListing.cs:1]: datetime

**Comment by JasonHomely** [comment]
> Also @JasonHomely - Codecov highlighted an interesting point -> sub and lot are not getting used anywhere?
> 
> the `Address` class has some methods for printing out the address. These need to be included IMO, in these methods.

added

File: src/OpenRealEstate.Core/Address.cs (hunk 1 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/Address.cs:1]: damn i sometimes hate PR view where it shows such a small section and you _think_ that's the entire file.

File: src/OpenRealEstate.Core/Residential/ResidentialListing.cs (hunk 8 above)
Comment (PureKrome) [reply src/OpenRealEstate.Core/Residential/ResidentialListing.cs:1]: I suck at this Pr reviewing stuff :)

**Review by PureKrome — APPROVED** [review]

**Pull request merged**